plugin-c/crypto.cpp

```diff
@@ -27,6 +28,30 @@ struct string *hook_KRYPTO_sha512(struct string *str) {
   return hexEncode(digest, sizeof(digest));
 }
 
+struct string *hook_KRYPTO_sha512_256raw(struct string *str) {
+  unsigned char digest[32];
+  unsigned int digest_len = 32;
+  EVP_MD_CTX *ctx = EVP_MD_CTX_new();
+  assert(ctx != NULL);
```

Comment:
you are using assert here in a way that is not a good idea. If we compile in release mode, these assertions will vanish, and this function will return a sequence of uninitialized bytes.